applications/tari_console_wallet/src/lib.rs

```diff
@@ -213,13 +215,16 @@ fn get_password(config: &ApplicationConfig, cli: &Cli) -> Option<SafePassword> {
 fn get_recovery_seed(boot_mode: WalletBoot, cli: &Cli) -> Result<Option<CipherSeed>, ExitError> {
     if matches!(boot_mode, WalletBoot::Recovery) {
         let seed = if cli.seed_words.is_some() {
-            let seed_words: Vec<String> = cli
-                .seed_words
-                .clone()
-                .unwrap()
-                .split_whitespace()
-                .map(|v| v.to_string())
-                .collect();
+            // need to zeroize first, to clean up memory of cli.seed_words clone
+            let vec_words = Zeroizing::new(cli.seed_words.clone());
+            let seed_words: SeedWords = SeedWords::new(
+                vec_words
+                    .as_ref()
+                    .unwrap()
+                    .split_whitespace()
+                    .map(|v| Hidden::hide(v.to_string()))
+                    .collect(),
+            );
```

Comment:
You could just move each string into the Hidden.
```suggestion
            let seed_words: SeedWords = SeedWords::new(
                cli.seed_words.clone()
                    .unwrap()
                    .split_whitespace()
                    .map(Hidden::hide)
                    .collect(),
            );
```

Note that cli.seed_words still leaks